6029 bug i datovelger

janitamarielle opened the pull request:
Datovelger returnerer onChange med norskString istedenfor Date. Dette for å støtte endring til reell verdi, og ikke bare gyldige verdier. 

Gjør det mulig å bruke intern valideringer for de elementene som ikke har egen støtte for feilmeldinger. 

Lager egen PR for omskriving av soknadsperiode til yup+react hook for å unngå hacky løsninger.


https://jira.adeo.no/browse/MELOSYS-6029

janitamarielle committed:
Justerer slik at verdi alltid blir oppdatert til faktisk verdi, og ikke bare når faktisk verdi er en gyldig Date. Fjerner også felles validering på selve datovelgeren

janitamarielle committed:
Gjør oppførsel likt slik det var for de uten typ-validering, og iallefall ikke verre. De som ikke har egen feil-håndtering bruker intern validering.
Alle som bruker den er endret fra Date til norskString ved onChange

janitamarielle committed:
Litt klarere hva som er parameter i onChange-funksjonen

janitamarielle committed:
Test

janitamarielle commented:
Egen PR for omskriving av soknadsperiode: https://github.com/navikt/melosys-web/pull/2024 Sikkert greit å ta #2024  først, siden den merges inn i denne

janitamarielle committed:
Fiks lint

janitamarielle committed:
Merge branch 'master' of https://github.com/navikt/melosys-web into MELOSYS-6029_bug-i-datovelger

IsaSalemHame commented on src/felleskomponenter/datovelger/datovelger.tsx:
```diff
@@ -39,20 +42,26 @@ const Datovelger = ({
+      onChange(Utils.dato.formatterDatoTilNorsk(nyValgtDatoFraDatePicker, false, undefined)),
     onValidate: (err) => {
-      if (err.isBefore || err.isAfter) {
+      if (!brukInternValidering) return;
```
Bra med option

janitamarielle replied:
Bra :D Blir ganske få som har optionet på etter https://github.com/navikt/melosys-web/pull/2024 da

IsaSalemHame reviewed (approved):
Skikkelig brainfuck oppgave, men ser ut til å funke bra nå! 

En vakker dag skal vi rydde litt i denne inceptionen av datovelger.

janitamarielle committed:
6029 Skriver om soknadsperiode til react hook form (#2024)

* Skriver om soknadsperiode til react hook form
soknadsperiode som realtime oppdaterte redux-mottatteopplysninger ble vel hacky etterhvert. Endret til å ha eget react hook form + schema for å gjøre det lett å se hva som faktisk skjer + å ha validering på faktisk verdi av dato og ikke verdi etter Utils-formattering til ISO.
La til snapshot tester

* Lint

* Snapshot-endringer kom ikke med

* uuid-greier

janitamarielle committed:
Merge branch 'master' of https://github.com/navikt/melosys-web into MELOSYS-6029_bug-i-datovelger

janitamarielle committed:
Liten bug der reset initialiserte til første verdier, selv om vi hadde endret én gang allerede.
